lib/src/r_tree/non_leaf_node.dart

```diff
@@ -124,16 +124,26 @@ class NonLeafNode<E> extends Node<E> {
     if (nonLeafParent == null) return;
 
     var newLeafNode = LeafNode<E>(this.branchFactor);
-    newLeafNode.include(this);
     nonLeafParent.removeChild(this);
     nonLeafParent.addChild(newLeafNode);
   }
 
-  _recalculateHeight() {
-    final maxChildHeight = _childNodes.fold(0, (int greatestHeight, childNode) {
-      return max(greatestHeight, childNode.height);
-    });
+  _updateHeightAndBounds() {
+    var height = 1;
+    var minimumBoundingRect = const Rectangle<num>(0, 0, 0, 0);
 
-    height = 1 + maxChildHeight;
+    if (children.isNotEmpty) {
+      height += _childNodes.fold(0, (int greatestHeight, childNode) {
+        return max(greatestHeight, childNode.height);
+      });
+
+      minimumBoundingRect = children.first.rect;
+      for (final child in children.skip(1)) {
+        minimumBoundingRect = minimumBoundingRect.boundingBox(child.rect);
+      }
```

Comment:
FYI I noticed the anonymous closure on fold here when profiling, we're iterating the children twice here and adding a closure to the stack ain't free. This seems about 5% faster for the remove benchmark:

```suggestion
      final firstChild = children.first;
      var maxChildHeight = firstChild.height;
      minimumBoundingRect = firstChild.rect;
      for (final child in children.skip(1)) {
        minimumBoundingRect = minimumBoundingRect.boundingBox(child.rect);
        if (child.height > maxChildHeight) {
          maxChildHeight = child.height;
        }
      }
      height += maxChildHeight;
```

Reply:
Hmm, that's good to know. I'll make this change.